src/Tokenizer/Token.php

```diff
@@ -125,52 +134,40 @@ public function equals($other, bool $caseSensitive = true): bool
         if ('&' === $other) {
             return '&' === $this->content && (null === $this->id || $this->isGivenKind([FCT::T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG, FCT::T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG]));
         }
-        if (null === $this->id && '&' === $this->content) {
+        if ('&' === $this->kind) {
             return $other instanceof self && '&' === $other->content && (null === $other->id || $other->isGivenKind([FCT::T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG, FCT::T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG]));
         }
 
         if ($other instanceof self) {
-            // Inlined getPrototype() on this very hot path.
             // We access the private properties of $other directly to save function call overhead.
             // This is only possible because $other is of the same class as `self`.
-            if (!$other->isArray) {
-                $otherPrototype = $other->content;
-            } else {
-                $otherPrototype = [
-                    $other->id,
-                    $other->content,
-                ];
-            }
-        } else {
-            $otherPrototype = $other;
+            $other = [$other->kind, $other->content];
```

Comment:
kind could be content, so it can possibly create `$other = [';', ';']` ?


```suggestion
// We access the private properties of $other directly to save function call overhead.
// This is only possible because $other is of the same class as `self`.
            $other = [$other->id, $other->content];
```